### DISCUSSION TITLE
Has anyone tried Implementing Breakout rooms ?

### DISCUSSION THREAD

**Discussion by prayaslashkari** [edumeet/edumeet#820] [Ideas]

**Comment by so010** [comment]
Not yet but it is on quite popular request from our users so we look at it and had some discussions about how to implement. User interface is quite important here.

**Comment by ClaudioAllocchio** [comment]
I add some proposed engineering ideas on the breakout room topic. Many users think at breakout rooms the way they are implented in Zoom architecture: the meeting moderator creates the rooms, and then moves / ask people to move into the rooms, the rooms are totally separated one from the other. I think this is a STRONG limitation to the idea, and the limitation just comes from the internal "central point of A/V mixing" which is in Zoom architecture. 
We use webRTC, and can implement a much better, flexible, useful architecture on the concept of "breakout rooms". 
We shall of course allow the same very basic functions described above, with the same principles. However we shall add to this:
- when breakout rooms exists, there is a central "atrium room" where ALL existing breakout rooms are visible. they look like a composite single video stream which is what you would see if you are in that room. You can see (by clicking some option on the room icon/video) who is in the room, and in case enable audio from the room you are clicking on. 
- the user can then decide autonomously to enter / ask to be admitted into a room he selects and he is in the room
- when the user leaves a room he goes back to the "atrium" room and can start again the process.

The moderator of the meeting of course has the power to configure the breakout sessions choosing the mode he prefers (usual Zoom style stuff, the new way I propose above). This is just a rough scenario. to be discussed.

the additional mode I propose will make eduMEET a very useful tool also to handle "coffee break style" discussions during an event/meeting. It will also be perfect to handle a "talk to the speaker" time slot, too.

**Comment by varundabke** [comment]
I have implemented one to one Breakout rooms ( not group ) and are very simple to use ( compared to zoom ).
I basically create a New room and a invitation to its participants pops up on their screen.

**Reply by JoseIsra** [reply]
how do you handle breakout rooms? i understand that you need to duplicate what you use in the main room for the breakoutroom because a breakout room is actually a copy from the main room, but how is this possible

**Comment by havfo** [comment]
In the new client and room-server breakout rooms are now available. You stay within the same overall room, but your media/chat is separate while in the breakout. Broadcasting to all breakouts is yet to be implemented.
